Вывод комментариев на экран

RomanNikolaevich opened the pull request:
Вывод комментариев с минимальным функционалом.
пагинатор пока не рабочий

RomanNikolaevich committed:
Вывод комментариев на экран

RomanNikolaevich committed:
Разнес комментарии на разные страницы для пагинатора и страница комментов сломалась

RomanNikolaevich committed:
Добавил проверку, что если пользователь залогинился, то в комментариях не нужно снова логин писать, будет подтягиваться автоматом

RomanNikolaevich committed:
Добавил проверку, что если пользователь залогинился, то в комментариях не нужно снова логин писать, будет подтягиваться автоматом

RomanNikolaevich committed:
проверка длинны комментариев, чтобы были не короче 50 символов

RomanNikolaevich committed:
доделал пагинатор
остались мелкие ошибки с логином - отправляю комментарий и кнопкалогин исчезает и появляется логаут

RomanNikolaevich committed:
Новая сборка по уроку 21
1) Для тех кто уже оставлял комментарий поле введения комментариев теперь не исчезает,
а теперь оно заблокировано и в нем указан 'username' комментатора.
2) Добавил кнопку "Перезайти" (логаут) в комментариях для смены 'username'.
3) Добавил функцию для длинных комментариев, чтобы если пользователь делает перенос текста,
то при выводе этот перенос сохранится.

Баги: комментарий, залогиненого через комментарии пользователя, не попадает в БД и выводится
на страницу.

RomanNikolaevich committed:
Новая сборка по уроку 21 = 17.05.2022 - 2
 Добавил к каждому посту в комментариях вывод № ID, но так как у меня один комментарий удален,
 то комментариев в реальности 26, а ID последний=27
 Добавил счетчик комментариев в обратном порядке, но он сбрасывается на каждой новой
 гет странице и счет снова начинается с 26.

JokerVDen commented on css/normalize.css:
```diff
@@ -302,12 +302,12 @@ textarea {
 ::-webkit-file-upload-button {
   -webkit-appearance: button; /* 1 */
-  font: inherit; /* 2 */
+  font: inherit; /* 2.php */
```
Зачем этот комментарий?

JokerVDen replied:
Везде в этом файле

RomanNikolaevich replied:
я тут ничего не вносил - это что-то сам IDE что-то там понаписал

RomanNikolaevich replied:
скачал новый нормалайз цсс, вместо старого

JokerVDen replied:
Не советую так делать. По причине, что это:
- может внести правки которые весь дизайн положат!!!!
- бессмысленны так как старая реализация работала как надо
- обычно таки изменения делаются очень редко, и весь сайт потом проверяется на недопустимые изменения

RomanNikolaevich replied:
принял

JokerVDen commented on modules/comments/main.php:
```diff
@@ -0,0 +1,64 @@
+                $username = $_SESSION['username'];
+            }
+            $comment = $_POST['comment'];
+            $query = "INSERT INTO comments SET name='$username', text='$comment'";
```
Две переменные что сюда прилетают должны быть экранированы. Если оставишь так, то твою бд могут поламать. Или почитай как можно экранировать данные, есть специальные функции или байнд для этого

RomanNikolaevich replied:
исправил

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+                    <?php
+                    if (empty($_SESSION['username'])) { ?> <!--блок коментатора-->
+						<input type="text" class="form-control" name="username" id="loginReg"
+							   value="<?=@htmlspecialchars($_POST['username']);?>"
```
Вот зачем здесь @? Ты же условие вверху поставил, тут пр  любых раскладах ошибки не будет. Давай так, забудь о @ , пользуйся условиями

RomanNikolaevich replied:
убрал улитку, выдало ошибку:
E:\INET\OpenServer\domains\testphpagro1.ua\skins\default\comments\main.tpl on line 27

Deprecated: htmlspecialchars(): Passing null to parameter #1 ($string) of type string is deprecated in E:\INET\OpenServer\domains\testphpagro1.ua\skins\default\comments\main.tpl on line 27
" placeholder="Введите логин *">

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+                    } else { ?> <!--заканчивается 1 блок условие "если не зарегистрирован"-->
+						<input type="text" class="form-control" name="disabledusername" id="disabledTextInput"
+							   value=""
+							   placeholder="<?php
```
Вот в этом месте сделай вывод только переменной из сессии, остальное тупо текстом спереди добавь. Используй коротку запись на вывод на экран <?=

RomanNikolaevich replied:
исправил

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+							# <?php if($commentCount > 0) {echo $commentCount--;} ?> |
+							post ID: <u><?=$comment['id']?></u> | user: <u><?=$comment['name']?></u> |
+							date:  <u><?=$comment['date']?></u> | : <br>
+							<i><?=str_replace( "\n", '<br />', $comment['text']); ?></i><br>
```
nl2br почитай о этой функции и заюзай

RomanNikolaevich replied:
85-ю строчку я как раз из https://www.php.net/manual/ru/function.nl2br.php взл из комментариев - там пишут, что str_replace( "\n", '<br />', $comment['text']); лучше использовать

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+                if ($pageno > 1): ?>
+					<li class="page-item">
+						<a class="page-link" href="?module=comments&page=
+							<?php
```
Короткая запись <=?

RomanNikolaevich replied:
исправил

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+                if ($pageno > 1): ?>
+					<li class="page-item">
+						<a class="page-link" href="?module=comments&page=<?php
+                        echo($pageno - 1); ?>">
```
Короткая запись

RomanNikolaevich replied:
исправил

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+				<li class="page-item">
+					<a class="page-link" href="?module=comments&page=<?php
+                    echo $pageno; ?>">
+                        <?php
```
Короткая запись

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+					<li class="page-item">
+						<a class="page-link" href="?module=comments&page=<?php
+                        echo($pageno + 1); ?>">
+                            <?php
```
Короткая запись

RomanNikolaevich replied:
исправил

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+			</ul>
+		</div>
+	</div>
+</div>
```
Нажми на кружок и почитай))

RomanNikolaevich replied:
у меня тут нет кружка никакого

JokerVDen commented on skins/default/comments/main.tpl:
```diff
@@ -0,0 +1,153 @@
+                if ($pageno < $total_pages): ?>
+					<li class="page-item">
+						<a class="page-link" href="?module=comments&page=<?php
+                        echo($pageno + 1); ?>">Next</a>
```
Короткая запись, проверь все echo в tpl файле и если можно сделай короткую запись

RomanNikolaevich replied:
исправил

RomanNikolaevich committed:
Новая сборка по уроку 21 = 18.05.2022 - 1
1) почистил код - короткую запись php сделал
2) добавил экранирование
3) улитки пришлось оставить, так как без них выдает ошибки

RomanNikolaevich committed:
Новая сборка по уроку 21 = 18.05.2022 - 1
1) почистил код - короткую запись php сделал
2) добавил экранирование
3) улитки пришлось оставить, так как без них выдает ошибки

RomanNikolaevich committed:
финальная сборка
